pkg/client/results/junit.go

```diff
@@ -18,25 +18,104 @@ package results
 
 import (
 	"encoding/xml"
+	"fmt"
+	"io"
 	"os"
 	"path/filepath"
 
-	"github.com/onsi/ginkgo/reporters"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
 const (
-	// JunitStdoutKey is the key in the Items.Details map for the system-out output.
-	JunitStdoutKey = "system-out"
+	// JUnitStdoutKey is the key in the Items.Details map for the system-out output.
+	JUnitStdoutKey = "system-out"
 
-	// JunitFailureKey is the key in the Items.Details map for the failure output.
-	JunitFailureKey = "failure"
+	// JUnitFailureKey is the key in the Items.Details map for the failure output.
+	JUnitFailureKey = "failure"
 )
 
-// Filter keeps only the tests that match the predicate function.
-func Filter(predicate func(testCase reporters.JUnitTestCase) bool, testSuite reporters.JUnitTestSuite) []reporters.JUnitTestCase {
-	out := make([]reporters.JUnitTestCase, 0)
+// junitResult is a wrapper around the suite[s] which enable results to
+// be either a single suite or a collection of suites. For instance,
+// e2e tests (which use the onsi/ginkgo reporter) report a single, top-level
+// testsuite whereas other tools report a top-level testsuites object
+// which may have 1+ testsuite children. Only one of the fields should be
+// set, not both.
+type junitResult struct {
+	suites JUnitTestSuites
+}
+
+// UnmarshalXML will unmarshal the document into either the 'Suites'
+// field of the JUnitResult. If a single testsuite is found (instead of a testsuites object)
+// it is unmarshalled and then added into the set of JUnitResult.Suites.Suites
+func (j *junitResult) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	var returnErr error
+	if start.Name.Local == "testsuites" {
+		returnErr = d.DecodeElement(&j.suites, &start)
+	} else {
+		var s JUnitTestSuite
+		returnErr = d.DecodeElement(&s, &start)
+		j.suites = JUnitTestSuites{Suites: []JUnitTestSuite{s}}
+	}
+	for i := range j.suites.Suites {
+		if j.suites.Suites[i].Name == "" {
+			j.suites.Suites[i].Name = fmt.Sprintf("testsuite-%03d", i+1)
```

Comment:
When processing junit results, each suite is on its own branch. As a result, if you dont have a suite name you get a weird branch that you can't target very well.

In particular, this happens for the e2e tests so I wanted to ensure that non-empty names were set.